src/dc_receive_imf.rs

```diff
@@ -407,6 +407,17 @@ async fn add_parts(
                 .await
                 .unwrap_or_default();
 
+        if chat_id.is_unset() {
+            // check if the message belongs to an NDN
+            if mime_parser.failed_msg.is_some() {
+                *chat_id = ChatId::new(DC_CHAT_ID_TRASH);
```

Comment:
we do not check here that the message-id is actually in use, we check that later, when the message is already thrown away, in handle_reports().

for real reports this is no issue, and probably currently also the heuristics are often strong enough, however, this may easily change in the future - and already today, eg. matt.deamon@gmx.de cannot send us a log with the subject "Failing to use Delta Chat" ;)

so maybe better check esp. for heuristics that the target message-id actually exists. we could check that in heuristically_parse_ndn() or here if heuristically_parse_ndn() sets a flag is_heuristic to the NDN.

Reply:
Done.